test/Microsoft.ML.Tests/DatabaseLoaderTests.cs

```diff
@@ -79,6 +80,59 @@ public void IrisLightGbm()
             }).PredictedLabel);
         }
 
+        [LightGBMFact]
+        public void IrisLightGbmCommandTimeout()
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                // https://github.com/dotnet/machinelearning/issues/4156
+                return;
+            }
+
+            var mlContext = new MLContext(seed: 1);
+
+            var connectionString = GetConnectionString(TestDatasets.irisDb.name);
+            var commandText = $@"WAITFOR DELAY '00:00:02'; SELECT * FROM ""{TestDatasets.irisDb.trainFilename}""";
+            var commandTimeout = 1;
+
+            var loaderColumns = new DatabaseLoader.Column[]
+            {
+                new DatabaseLoader.Column() { Name = "Label", Type = DbType.Int32 },
+                new DatabaseLoader.Column() { Name = "SepalLength", Type = DbType.Single },
+                new DatabaseLoader.Column() { Name = "SepalWidth", Type = DbType.Single },
+                new DatabaseLoader.Column() { Name = "PetalLength", Type = DbType.Single },
+                new DatabaseLoader.Column() { Name = "PetalWidth", Type = DbType.Single }
+            };
+
+            var loader = mlContext.Data.CreateDatabaseLoader(loaderColumns);
+
+            var databaseSource = new DatabaseSource(SqlClientFactory.Instance, connectionString, commandText, commandTimeout);
+
+            bool passed=false;
+
+            try
+            {
+                var trainingData = loader.Load(databaseSource);
+
+                IEstimator<ITransformer> pipeline = mlContext.Transforms.Conversion.MapValueToKey("Label")
+                .Append(mlContext.Transforms.Concatenate("Features", "SepalLength", "SepalWidth", "PetalLength", "PetalWidth"))
+                .AppendCacheCheckpoint(mlContext)
+                .Append(mlContext.MulticlassClassification.Trainers.LightGbm())
+                .Append(mlContext.Transforms.Conversion.MapKeyToValue("PredictedLabel"));
+
+                var model = pipeline.Fit(trainingData);
+            }
+            catch(Exception e)
```

Comment:
You can use `Assert.Throws` instead.

Also - you should ensure the correct type of exception is thrown.